### PR TITLE
doc: announce C++11 changes

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8735)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,13 @@
 This repository contains idiomatic C++ client libraries for the following
 [Google Cloud Platform](https://cloud.google.com/) services.
 
+> Please check the [CHANGELOG] for important announcements and upcoming
+> changes.<br>
+> In particular, note that on (or about) **2022-06-01** this library will stop
+> development based on C++11.
```
Comment (devjgm) [inline README.md:34]: We may want to make it clear that the new baseline will be C++14, which may not be obvious. What about saying "... this library will start requiring >= C++14"?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,18 @@
 
 <!-- Keep these sorted by estimated date -->
 
+<details>
+<summary>2022-06-01: last release with C++ support</summary>
```
[the remark this answers is not in the document]
Comment (coryan) [reply CHANGELOG.md]: Fixed

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,18 @@
...
+
+* The release on 2022-06-01 (tentatively v1.41.0) will be the last release
+  supporting C++11. This we plan to backport critical (P0) bugs and security
```
[the remark this answers is not in the document]
Comment (coryan) [reply CHANGELOG.md]: Fixed

File: README.md (hunk 1 above)
Comment (coryan) [reply README.md:34]: Moved earlier in the text.

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,18 @@
...
 <!-- Keep these sorted by estimated date -->
 
+<details>
+<summary>2022-06-01: last release with C++11 support</summary>
+
+* The release on 2022-06-01 (tentatively v1.41.0) will be the last release
+  supporting C++11, future releases will require C++ >= 14. We plan to backport
+  critical (P0) bugs and security fixes to this release for a year, that is,
+  until 2023-06-01.
+  The release following will be v2.1.0, to more clearly communicate this change.
```
Comment (devbww) [inline CHANGELOG.md:14]: `v2.0.0` perhaps?

Comment (coryan) [reply CHANGELOG.md]: Fixed

File: README.md (hunk 1 above)
Comment (devjgm) [reply README.md:34]: JFYI, I don't see the fix, but I'm also happy to consider my comment optional if you'd prefer to leave the text as it is now.

**Review by devjgm — APPROVED** [review]

**Review by devjgm — APPROVED** [review]

**Pull request merged**